tokio/src/net/addr.rs

```diff
@@ -40,6 +40,61 @@ where
     }
 }
 
+cfg_dns! {
+    /// LookupHost contains resolved SocketAddrs.
+    ///
+    /// This type is created by the function [`lookup_host`]. See its documentation for more.
+    #[derive(Debug)]
+    pub struct LookupHost<T>
+    where
+        T: sealed::ToSocketAddrsPriv,
+        T: ToSocketAddrs<Future = sealed::MaybeReady>,
+        T: ToSocketAddrs<Iter = sealed::OneOrMore>,
+    {
+        fut: <T as sealed::ToSocketAddrsPriv>::Future,
+        addrs: Option<sealed::OneOrMore>
+    }
+
+    impl<T> LookupHost<T>
+    where
+        T: sealed::ToSocketAddrsPriv + Unpin,
+        T: ToSocketAddrs<Future = sealed::MaybeReady>,
+        T: ToSocketAddrs<Iter = sealed::OneOrMore>,
+    {
+        /// Fetches the next resolved address.
+        pub async fn next_addr(&mut self) -> Option<io::Result<SocketAddr>> {
+            if self.addrs.is_none() {
+                let hosts = &mut self.fut;
+                    let hosts = hosts.await;
+                    match hosts {
+                        Ok(addrs) => self.addrs = Some(addrs),
+                        Err(e) => return Some(Result::Err(e))
+                };
+            }
+            let addrs = &mut self.addrs;
+            if let Some(addrs) = addrs {
+                let next = addrs.next();
+                next.map(Ok)
+            } else {
+                unreachable!()
+            }
+        }
+    }
+
+    /// Performs a DNS resolution.
+    ///
+    /// This API is not intended to cover all DNS use cases.
```

Comment:
Could you add some examples?

When is `Err` returned, when is `None` returned? Is it guaranteed to return a value or can it be "empty"?

Reply:
I think the current revision documents the behavior.